CMakeLists.txt

```diff
@@ -1,21 +1,77 @@
+
+###--------------------------------------------------------------------------------------------------- 
+#[[ This part is intended to custom targets5 and activates the custom_targets.json5 
+and upload method config otherwise functions below should be commented ]]
+
+#[[ Here set path for custom_targets.json5 (this is our default) ]]
+ set(CUSTOM_TARGETS_JSON_PATH custom_targets/custom_targets.json5) 
+
+#[[ Here you can set path for custom upload config .cmake (optional example) ]]
+# set(CUSTOM_UPLOAD_CFG_PATH ${CMAKE_SOURCE_DIR}/${MBED_TARGET}/${MBED_TARGET}.cmake)
+
+#[[ Note: For custom target you need also an upload method and we have few options how you can do that
+    - use the variable CUSTOM_UPLOAD_CFG_PATH below
+    - use default expected path for custom targets upload methods where you create your own config
+      MY_PROJECT/custom_targets/upload_method_cfg
+    - of course you can do it by your self dierectly via cmake in this file
+
+    For regular targets we use mbed-os/tree/master/targets/upload_method_cfg ]]
+
+### End of block
+######################################################################################################
+
+### include app.cmake (necessary everytime) ###
 include(mbed-os/tools/cmake/app.cmake)
 
-# Include folder with the custom targets.
-# This needs to be done before add_subdirectory(mbed-os).
+######################################################################################################
+### Block of including project folders
+
+#[[ In case of custom targets must be the folder included before mbed-os subdirectory
+otherwise should be the next line commented]]
 add_subdirectory(custom_targets)
 
+###--------------------------------------------------------------------------------------------------
+## Add mbed-os subdirectory (necessary everytime)
 add_subdirectory(mbed-os)
 
+###--------------------------------------------------------------------------------------------------
+## Add another subdirectory, for example subdirectory of a library (if needed)
+#add_subdirectory(YourLibrary)
+
+### End of block
+######################################################################################################
+
+### Set up your project name (necessary everytime)
 project(mbed-ce-custom-targets)
 
-# add subdirectories and build targets here
 
+### add executable (necessary everytime)
+add_executable(main main.cpp)
+
+######################################################################################################
+### Link libraries block
+#[[For more about this configuraion visit wiki page MbedOS-configuration ]]
+
+#[[link MbedOS and its libraries (necessary everytime)]]
+target_link_libraries(main  mbed-os) 
+
+### link user library (if needed)
+#target_link_libraries(main YourLibrary) 
+
+### End of block
+######################################################################################################
+
+### Set post build (necessary everytime)
+mbed_set_post_build(main)
```

Comment:
Nitpick: can we move this up below the `add_executable()` call?  Really each `add_executable()` should be paired up with an `mbed_set_post_build()` call.  Note that it's perfectly ok to call mbed_set_post_build() before target_link_libraries().

Reply:
As you wish.